addons/web/static/src/legacy/action_adapters.js

```diff
@@ -360,3 +345,11 @@ export class ViewAdapter extends ActionAdapter {
         }
     }
 }
+
+export function handleDoAction(actionService, eventPayload) {
```

Comment:
I think I wouldn't try to factorize the code here. It adds unnecessary indirection and complexity imho. In the legacy provider case, the implementation is simpler (no need to apply the wrapSuccessOfFail stuff)